mindsdb_native/libs/data_types/transaction_output_row.py

```diff
@@ -45,23 +45,19 @@ def explain(self):
             else:
                 answers[pred_col]['predicted_value'] = prediction_row[pred_col]
 
-
-            if f'{pred_col}_model_confidence' in prediction_row:
-                answers[pred_col]['confidence'] = (prediction_row[f'{pred_col}_model_confidence'] * 3 + prediction_row[f'{pred_col}_confidence'] * 1)/4
```

Comment:
This will affect confidence quite heavily, since no we are weighting just nonconformist whereas before, at least judging by this, the non-conformist confidence weight was just 1/4, the rest being the selfaware weight (model_confidence)

Or am I wrong here ?

Reply:
I may well be

Reply:
No, you are right in that it affects heavily. I did some benchmarks which indicate it's an improvement (7.7% over 13 datasets), but we could revert it back for the time being and benchmark further (disclaimer: I only tested numerical tasks).

Reply:
Did some more testing, it definitely seems like an improvement.